nexus/src/app/background/init.rs

```diff
@@ -739,9 +740,39 @@ pub mod test {
     use nexus_types::internal_api::params as nexus_params;
     use omicron_test_utils::dev::poll;
     use std::net::SocketAddr;
+    use std::sync::atomic::AtomicU64;
+    use std::sync::atomic::Ordering;
     use std::time::Duration;
     use tempfile::TempDir;
 
+    /// Used by various tests of tasks that kick off sagas
```

Comment:
It might be nice, eventually, to also be able to make assertions about which sagas were started and what params they were started with, but that seems like something we can add later...

Reply:
Agreed.  This starts getting tricky for both accidental reasons (the SagaDag doesn't expose everything we'd probably like here) and intrinsic reasons (the params structs are saga-specific).  We could specialize this struct to specific sagas and have it keep a queue of params (and I started down that path before I noticed we weren't actually checking the values of the params), but one of these tasks actually starts two different sagas.